### ISSUE TITLE
Aim for no warnings

### ISSUE THREAD

**Issue by dorchard** [dorchard/granule#8]
I noticed that we have a lot of warnings popping up when compiling (mostly things like unused variables or unnecessary imports). We should tidy these up (perhaps a useful code exploration activity @vimuel ?)

**Comment by vimuel** [comment]
Yep, I was planning on doing this as well. :)

**Comment by vimuel** [comment]
We now only have 5 orphan instance warnings left in the module `Checker.Types`.

What shall we do about these?

**Comment by vimuel** [comment]
I have just found another warning, which I'm not sure what to do about: 
```
gram-lang/src/Checker/Environment.hs: 76, 10
• No explicit implementation for
    ‘local’
• In the instance declaration for ‘MonadReader [(Id, Id)] Checker’
```

**Comment by vimuel** [comment]
Oh and then there is this:
```
Warning: The following modules should be added to exposed-modules or other-modules in /Users/vilem/code/github/gram-lang/gram.cabal:
    - In spec component:
        Checker.Types
        Context
        Syntax.Expr
        Syntax.Pretty
```
which I find strange, because they are listed under `Other-modules` in `gram.cabal`.

**Comment by dorchard** [comment]
This was fixed by 9c6d3bda96ea46e39e61eab6a9a7120bb1e53f36
We still have some orphan instances warnings, which we can get rid of at some point; I'll close this now and let's keep the general principle of: aim for no warnings.
